src/DynamicNLPModels.jl

```diff
@@ -196,18 +196,67 @@ end
 
 abstract type AbstractDynamicModel{T,V} <: QuadraticModels.AbstractQuadraticModel{T, V} end
 
-mutable struct LQDynamicModel{T, V, M1, M2, M3, MK} <:  AbstractDynamicModel{T,V} 
+mutable struct SparseLQDynamicModel{T, V, M1, M2, M3, MK} <:  AbstractDynamicModel{T,V} 
   meta::NLPModels.NLPModelMeta{T, V}
   counters::NLPModels.Counters
   data::QuadraticModels.QPData{T, V, M1, M2}
   dynamic_data::LQDynamicData{T, V, M3, MK}
-  condense::Bool
+end
+
+mutable struct block_matrices{T, M1, M2}
```

Comment:
Does it have to be mutable?

Reply:
I don't think it needs to be mutable. I can change it to be immutable

Reply:
type names should be capitalized